integration-tests/jms-ibmmq-client/src/test/java/org/apache/camel/quarkus/component/jms/ibmmq/it/IBMMQTest.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.quarkus.component.jms.ibmmq.it;
+
+import java.lang.reflect.Method;
+
+import io.quarkus.test.common.QuarkusTestResource;
+import io.quarkus.test.junit.QuarkusTest;
+import io.restassured.RestAssured;
+import org.apache.camel.quarkus.messaging.jms.AbstractJmsMessagingTest;
+import org.apache.camel.quarkus.test.support.ibmmq.IBMMQDestinations;
+import org.apache.camel.quarkus.test.support.ibmmq.IBMMQTestResource;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInfo;
+import org.junit.jupiter.api.TestInstance;
+import org.junit.jupiter.api.condition.EnabledIfSystemProperty;
+
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.startsWith;
+
+@QuarkusTest
+@QuarkusTestResource(IBMMQTestResource.class)
+@EnabledIfSystemProperty(named = "ibm.mq.container.license", matches = "accept")
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+public class IBMMQTest extends AbstractJmsMessagingTest {
+    IBMMQDestinations destinations;
+
+    /**
+     * IBM MQ needs to have the destinations created before you can use them.
+     * <p>
+     * This method is called after the routes start, so the routes will print a warning first that the destinations don't
+     * exist, only then they are
+     * created using this method
+     *
+     * @param test test
+     */
+    @BeforeAll
+    public void createDestinations(TestInfo test) {
+        for (Method method : test.getTestClass().get().getMethods()) {
+            destinations.createQueue(method.getName());
+            // Some tests use two queues
+            destinations.createQueue(method.getName() + "2");
+            destinations.createTopic(method.getName());
+        }
+
+        // Sleep for a while to allow the routes to reconnect
+        try {
+            Thread.sleep(5000L);
```

Comment:
Can't we perhaps create the queues and topics in the `IBMMQTestResource`? That one is started before the application under test, so we'd perhaps could avoid this reconnection interval?

Reply:
:thinking: is there a way how to access some test info in a test resource? that was my initial attempt, but I couldn't find a way how to do that, other than just hardcoding all the destinations it should create (which isn't very maintainable)

Reply:
Is there any particular reason to use method name as queue and topic?

Reply:
I'd vote for using names having some random suffix. You could create those in the `IBMMQTestResource` and pass them to the application under test via config properties returned from IBMMQTestResource.start().

Reply:
@ppalaga could you please elaborate? Wouldn't that require maintaining a list of destination names that should be created inside `IBMMQTestResource` when a new test is added to, for example, [AbstractMessagingTest](https://github.com/avano/camel-quarkus/blob/ibm-mq/integration-tests/messaging/common/src/test/java/org/apache/camel/quarkus/component/messaging/it/AbstractMessagingTest.java)?

Or did you mean to re-use one queue/topic for all tests?

Reply:
Yes, IBMMQTestResource would have to decide about the actual (random) names of the resources. The property names can be fixed in some common enum. I'd personally be for going that way rather than creating the resources in the test classes and having to wait for reconnecting. I guess the maintenance overhead will be smaller than the overall time lost due to waiting for reconnects. 

But maybe the problem with waiting for reconnect can be solved somehow. E.g. you do not auto-start the routes and instead you start them via REST from the test after you have created the required queues?

Reply:
> But maybe the problem with waiting for reconnect can be solved somehow. E.g. you do not auto-start the routes and instead you start them via REST from the test after you have created the required queues?

:+1: that should work, I'll look into it